Source/WebCore/html/canvas/CanvasRenderingContext2DBase.cpp

```diff
@@ -2222,7 +2227,38 @@ ExceptionOr<Ref<ImageData>> CanvasRenderingContext2DBase::getImageData(int sx, i
 
     ASSERT(pixelBuffer->format().colorSpace == toDestinationColorSpace(computedColorSpace));
 
-    return { { ImageData::create(static_reference_cast<ByteArrayPixelBuffer>(pixelBuffer.releaseNonNull())) } };
+    auto imageData = ImageData::create(static_reference_cast<ByteArrayPixelBuffer>(pixelBuffer.releaseNonNull()));
+    
+#if ENABLE(NOISY_APIS)
+    if (DeprecatedGlobalSettings::noisyAPIsEnabled()) {
+        IdentifierProvider* identifierProvider = canvasBase().scriptExecutionContext()->identifierProvider();
+        // Using WeakRandom as I could explicitly set the seed.
+        // FIXME: set and use canvasIdentifier instead of domainIdentifier.
+        WeakRandom randomNumberGenerator = WeakRandom(identifierProvider->domainIdentifier());
```

Comment:
Why aren't we using the `canvasIdentifier` here? (or rather, what is the purpose of the `canvasIdentifier`)